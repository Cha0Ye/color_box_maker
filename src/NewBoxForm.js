import React, { Component } from 'react';

class NewBoxForm extends Component {
    constructor(props){
        super(props)
        this.state = {
            width: "",
            height: "",
            backgroundColor: ""
        };
        this.handleChange = this.handleChange.bind(this);
    }

    handleChange(evt){

        this.setState({ 
            [evt.target.name]: evt.target.value
        });
    }

    handleSubmit(evt){
        evt.preventDefault();
        this.props.triggerCreate(this.state);
        this.setState({ width: "",
                        height: "",
                        backgroundColor: ""}) ;

    }

    render() {
        return (
            <div>
                <form onSubmit={ this.handleSubmit }>

                    <label htmlFor='width' >width: </label>
                    <input id='width' name='width' value={ this.state.width } onClick={ this.handleChange }> </input>
                    
                    <label htmlFor='height' >Height: </label>
                    <input id='height' name='height' value={ this.state.height } onClick={ this.handleChange }> </input>

                    <label htmlFor='backgroundColor' >Background Color: </label>
                    <input id='backgroundColor' name='backgroundColor' value={ this.state.backgroundColor } onClick={ this.handleChange }> </input>
                    <button>CreateBox</button>
                </form>
            </div>
        );
    }
}

export default NewBoxForm;