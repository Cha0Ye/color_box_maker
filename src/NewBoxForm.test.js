import React from 'react';
import NewBoxForm from './NewBoxForm';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

let wrapper;

beforeEach(()=>{
    wrapper = mount(<NewBoxForm />);
})

// smoke test
it('renders without crashing', () => {
  const wrapper = shallow(<NewBoxForm />);
});

// snapshot test
it('matches snapshot', () => {
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});

it("allows for changes in name, qty", function() {

    let wrapper = mount(<NewBoxForm />);
    const widthInput = wrapper.find("#width")
    widthInput.instance().value = "20px"
    widthInput.simulate("change")

    expect(wrapper.state().width).toEqual("20px");

    const heightInput = wrapper.find("#height");
    heightInput.instance().value = '20px';
    heightInput.simulate("change");

    expect(wrapper.state().height).toEqual('20px');

    const backgroundColorInput = wrapper.find("#backgroundColor");
    backgroundColorInput.instance().value = 'blue';
    backgroundColorInput.simulate("change");

    expect(wrapper.state().backgroundColor).toEqual('blue');

  });

  it("runs a mocked fn on submit", function () {
    const submitFn = jest.fn();
    let wrapper = mount(
      <NewBoxForm triggerCreate={submitFn} />
    );
    const form = wrapper.find("form")

    form.simulate("submit")

    expect(submitFn).toHaveBeenCalled();
  });

  it("resets state on submit", function () {
    const submitFn = jest.fn();

    let wrapper = mount(<NewBoxForm triggerCreate={submitFn} />);
    const widthInput = wrapper.find("#width")
    widthInput.instance().value = "20px"
    widthInput.simulate("change")

    expect(wrapper.state().width).toEqual("20px");

    const heightInput = wrapper.find("#height");
    heightInput.instance().value = '20px';
    heightInput.simulate("change");

    expect(wrapper.state().height).toEqual('20px');

    const backgroundColorInput = wrapper.find("#backgroundColor");
    backgroundColorInput.instance().value = 'blue';
    backgroundColorInput.simulate("change");

    expect(wrapper.state().backgroundColor).toEqual('blue');

    const form = wrapper.find("form");
    form.simulate("submit");

    // after submit, we expect the state to reset
    expect(wrapper.state()).toEqual({
      width: '',
      height: '',
      backgroundColor: '',
    })
  });

