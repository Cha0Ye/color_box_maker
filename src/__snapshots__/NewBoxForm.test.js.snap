// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<NewBoxForm>
  <div>
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="width"
      >
        width: 
      </label>
      <input
        id="width"
        name="width"
        onChange={[Function]}
        value=""
      />
      <label
        htmlFor="height"
      >
        Height: 
      </label>
      <input
        id="height"
        name="height"
        onChange={[Function]}
        value=""
      />
      <label
        htmlFor="backgroundColor"
      >
        Background Color: 
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        onChange={[Function]}
        value=""
      />
      <button>
        CreateBox
      </button>
    </form>
  </div>
</NewBoxForm>
`;
